extensions/azurecore/src/azureResource/azure-resource.d.ts

```diff
@@ -27,7 +27,8 @@ declare module 'azureResource' {
 			postgresServer = 'microsoft.dbforpostgresql/servers',
 			azureArcService = 'microsoft.azuredata/datacontrollers',
 			storageAccount = 'microsoft.storage/storageaccounts',
-			logAnalytics = 'microsoft.operationalinsights/workspaces'
+			logAnalytics = 'microsoft.operationalinsights/workspaces',
+			databaseMigrationService = 'microsoft.datamigration/sqlmigrationservices'
```

Comment:
This change might not be necessary, especially since it belongs to another extension. My intention with this change was to add `databaseMigrationService` as a new resource type (since `storageAccount` and `sqlManagedInstance` are already resource types) so that I can have the nice switch statement that you see below. If this change is too invasive then there are other ways to do this too

Reply:
Defined resources locally instead of in azurecore.